c_emulator/riscv_sim.cpp

```diff
@@ -532,23 +447,40 @@ void init_sail_reset_vector(uint64_t entry)
   for (uint64_t i = addr; i < rom_end; i++)
     write_mem(addr++, 0);
 
-  /* set rom size */
-  rv_rom_size = rom_end - rv_rom_base;
+  /* calculate rom size */
+  uint64_t rom_size = rom_end - rom_base;
+
+  /* check calculated rom values match configuration */
+  if (rom_base != get_config_uint64({"platform", "rom", "base"})) {
+    fprintf(stderr,
+            "Configuration value platform.rom.base does not match %" PRIu64
+            ".\n",
+            rom_base);
+  }
+  if (rom_size != get_config_uint64({"platform", "rom", "size"})) {
+    fprintf(stderr,
+            "Configuration value platform.rom.size does not match %" PRIu64
+            ".\n",
+            rom_size);
+  }
+
   /* boot at reset vector */
-  zPC = rv_rom_base;
+  zPC = rom_base;
 }
 
 void init_sail(uint64_t elf_entry)
 {
   zinit_model(UNIT);
   if (rvfi) {
+    /*
```

Comment:
Looks like this breaks RVFI, do we need a separate config file? Or can we override the sail variables here?

Reply:
You would need a separate configuration file with these values.

Reply:
I think if RVFI needs specific values it's probably not unreasonable for it to need a special config file. I think it's fine to leave this for now.